packages/react/menu/src/Menu.tsx

```diff
@@ -94,65 +237,179 @@ const MenuContent = React.forwardRef((props, forwardedRef) => {
-     * (default: `false`)
-     */
-    disableOutsideScroll?: boolean;
+  return (
+    <MenuContentImpl
+      {...props}
+      ref={composedRefs}
+      onDismiss={() => context.onOpenChange(false)}
+      onEntryFocus={(event) => event.preventDefault()}
+      onOpenAutoFocus={composeEventHandlers(props.onOpenAutoFocus, (event) => {
+        // explicitly focus the content area only when opening
+        event.preventDefault();
+        ref.current?.focus();
+      })}
+    />
+  );
+}) as MenuRootContentPrimitive;
 
-    /**
-     * The direction of navigation between menu items.
-     * @defaultValue ltr
-     */
-    dir?: RovingFocusGroupOwnProps['dir'];
+/* ---------------------------------------------------------------------------------------------- */
 
-    /**
-     * Whether keyboard navigation should loop around
-     * @defaultValue false
-     */
-    loop?: RovingFocusGroupOwnProps['loop'];
+type MenuSubContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  | keyof MenuContentImplPrivateProps
+  | 'side'
+  | 'portalled'
+  | 'disabledOutsidePointerEvents'
+  | 'disableOutsideScroll'
+  | 'trapFocus'
+  | 'onOpenAutoFocus'
+  | 'onCloseAutoFocus'
+>;
 
-    /**
-     * Whether the `MenuContent` should render in a `Portal`
-     * (default: `true`)
-     */
-    portalled?: boolean;
-  }
+type MenuSubContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuSubContentOwnProps
+>;
+
+const MenuSubContent = React.forwardRef((props, forwardedRef) => {
+  const context = useMenuContext(CONTENT_NAME);
+  const ref = React.useRef<MenuContentElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  return context.isSubmenu ? (
+    <MenuContentImpl
+      align="start"
+      id={context.contentId}
+      {...props}
+      ref={composedRefs}
+      side={context.dir === 'rtl' ? 'left' : 'right'}
+      portalled
+      disableOutsidePointerEvents={false}
+      disableOutsideScroll={false}
+      trapFocus={false}
+      onEntryFocus={context.onEntryFocus}
+      onOpenAutoFocus={(event) => event.preventDefault()}
+      // The menu might close because of focusing another menu item in the parent menu. We
+      // don't want it to refocus the trigger in that case so we handle trigger focus ourselves.
+      onCloseAutoFocus={(event) => event.preventDefault()}
+      onFocusOutside={composeEventHandlers(props.onFocusOutside, (event) => {
+        // We prevent closing when the trigger is focused to avoid triggering a re-open animation
+        // on pointer interaction.
+        if (event.target !== context.trigger) context.onOpenChange(false);
+      })}
+      onEscapeKeyDown={composeEventHandlers(props.onEscapeKeyDown, () => {
+        // We need to explicitly close when escape key focuses trigger because we prevented this
+        // in on focus outside for pointers.
```

Comment:
I find the comment reads weird here.

```suggestion
        // We need to explicitly focus the trigger when pressing escape because we prevented this
        // in `onCloseAutoFocus` so it doesn't happen with pointers.
```

Does this make sense?